plugins/search-react/src/components/SearchResultGroup/SearchResultGroup.stories.tsx

```diff
@@ -0,0 +1,234 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ComponentType, useCallback, useState } from 'react';
+
+import {
+  Divider,
+  Grid,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+} from '@material-ui/core';
+import DocsIcon from '@material-ui/icons/InsertDriveFile';
+
+import { TestApiProvider, wrapInTestApp } from '@backstage/test-utils';
+import { JsonValue } from '@backstage/types';
+import { createRouteRef } from '@backstage/core-plugin-api';
+import { CatalogIcon, Link } from '@backstage/core-components';
+import { SearchQuery } from '@backstage/plugin-search-common';
+
+import { searchApiRef, MockSearchApi } from '../../api';
+
+import {
+  SearchResultGroup,
+  SearchResultGroupTextFilterField,
+  SearchResultGroupSelectFilterField,
+} from './SearchResultGroup';
+
+const routeRef = createRouteRef({
+  id: 'storybook.test-route',
+});
+
+const searchApiMock = new MockSearchApi({
+  results: [
+    {
+      type: 'techdocs',
+      document: {
+        location: 'search/search-result1',
+        title: 'Search Result 1',
+        text: 'Some text from the search result 1',
+      },
+    },
+    {
+      type: 'techdocs',
+      document: {
+        location: 'search/search-result2',
+        title: 'Search Result 2',
+        text: 'Some text from the search result 2',
+      },
+    },
+  ],
+});
+
+export default {
+  title: 'Plugins/Search/SearchResultGroup',
+  component: SearchResultGroup,
+  decorators: [
+    (Story: ComponentType<{}>) =>
+      wrapInTestApp(
+        <TestApiProvider apis={[[searchApiRef, searchApiMock]]}>
+          <Grid container direction="row">
+            <Grid item xs={12}>
+              <Story />
+            </Grid>
+          </Grid>
+        </TestApiProvider>,
+        { mountedRoutes: { '/hello': routeRef } },
+      ),
+  ],
+};
+
+export const Default = () => {
+  const [query] = useState<Partial<SearchQuery>>({
+    types: ['techdocs'],
+  });
+
+  return (
+    <SearchResultGroup
+      icon={<DocsIcon />}
+      title="Documentation"
+      query={query}
+    />
+  );
+};
+
+export const WithNoResults = () => {
+  const [query] = useState<Partial<SearchQuery>>({
+    types: ['techdocs'],
+  });
+
+  return (
+    <TestApiProvider apis={[[searchApiRef, new MockSearchApi()]]}>
+      <SearchResultGroup
+        icon={<DocsIcon />}
+        title="Documentation"
+        query={query}
+        noResultItemsText="No results found"
+      />
+    </TestApiProvider>
+  );
+};
+
+// This is a simplified component only used as example, import it from the catalog plugin in production:
+// import { CatalogSearchResultListItem } from '@backstage/plugin-catalog';
+const CatalogSearchResultListItem = (props: any) => {
+  const { icon, result } = props;
```

Comment:
Hmm, I'm not sure I understand why we want to create this one instead of using `DefaultResultListItem` instead?

Reply:
I removed this simplified version of a `CatalogResulListItem` and used a `CustomResultListItem` instead 🙌🏻